index.html

```diff
@@ -2895,7 +2895,7 @@ <h2>
             </td>
             <td>
               <a data-cite="html/dom.html#flow-content">Flow
-              content</a><a data-cite="html/dom.html#phrasing-content-2"></a>,
+              content</a>,
```

Comment:
maybe this is more the result of a partial merge, rather than an errant empty link?

for instance, being in the restricted column, flow content is not allowed as a child of button, nor is phrasing content that is interactive (which could be what the intent was of the link removed?)

it's a bit hard to distinguish because the expected child content of a button should be:
>Phrasing content, but there must be no interactive content descendant.

Reply:
Yes, for an HTML button element that is true but is the same rule applied when we assign a role="button" to an element? 
Currently the Nu Html Checker provides two different results for the following:

- `<button type="button"><h1>hello</h1></button>` This is flagged as an error
- `<div role="button"><h1>hello</h1></div>` This is not flagged as an error

I was using the link and menuitem roles for comparison but that may not be the best approach.

Reply:
Seems like an oversight of the validator, rather than an example of why it's OK.

Reply:
Beyond button, this request also removes the empty links found in checkbox, dialog and heading. Is there a problem with removing these from those 3 roles?

Reply:
I reviewed and it looks like these empty links were included in my initial commit https://github.com/w3c/html-aria/commit/75f41b64ed16308e56da5e85cad8f7d5820c8c35, they are not supposed to be there so fine to remove

Reply:
@pgrucza @scottaohara is correct in that if an element with an implicit aria role permits only certain types of content, this should be generally reflected in the permissions for the role, although it is not one to one as the reason why people use aria roles rather than native elements, sometimes, is so they can do stuff that results in better styling, in the example case it is not a good idea either way to have a heading nested in a button, so the restrictions need to be tightened.